string target
---
bool target_switch_is_success